### ISSUE TITLE
snowflake_user unable to specify "default_secondary_roles" - fails on `apply`

### ISSUE THREAD

**Issue by dgniewek** [Snowflake-Labs/terraform-provider-snowflake#1579]
**Provider Version**  - v0.56.5

**Terraform Version** - v1.3.6

**Describe the bug**

When trying to pass `default_secondary_roles`  list to `snowflake_user`, `terraform plan` looks as expected, but `terraform apply` fails with:  

```shell
│ Error: error creating user err = 001008 (22023): SQL compilation error:
│ invalid value [[SNOWFLAKE_SECOND_ROLE]] for parameter 'DEFAULT_SECONDARY_ROLES'
```
Looks like an additional cast to list is somehow performed when creating snowflake SQL query that creates/modifies the user.

**Expected behavior**

`default_secondary_roles` for `snowflake_user` are successfully added.

**Code samples and commands**

Please add code examples and commands that were run to cause the problem.

Terraform code:
```terraform
resource "snowflake_role" "user_role" {
  name = "SNOWFLAKE_USER_ROLE"
}

resource "snowflake_role" "secondary_role" {
  name = "SNOWFLAKE_SECOND_ROLE"
}

resource "snowflake_user" "this" {
  count = module.this.enabled ? 1 : 0

  name = "TEST_USER"

  default_role            = resource.snowflake_role.user_role.name
  default_secondary_roles = [resource.snowflake_role.secondary_role.name]
}
```

Output:
```shell
$ terraform init 
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- Reusing previous version of snowflake-labs/snowflake from the dependency lock file
- Using previously-installed snowflake-labs/snowflake v0.56.5

Terraform has made some changes to the provider dependency selections recorded
in the .terraform.lock.hcl file. Review those changes and commit them to your
version control system if they represent changes you intended to make.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

$ terraform plan 

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # snowflake_role.secondary_role will be created
  + resource "snowflake_role" "secondary_role" {
      + id   = (known after apply)
      + name = "SNOWFLAKE_SECOND_ROLE"
    }

  # snowflake_role.user_role will be created
  + resource "snowflake_role" "user_role" {
      + id   = (known after apply)
      + name = "SNOWFLAKE_USER_ROLE"
    }

  # snowflake_user.this[0] will be created
  + resource "snowflake_user" "this" {
      + default_role            = "SNOWFLAKE_USER_ROLE"
      + default_secondary_roles = [
          + "SNOWFLAKE_SECOND_ROLE",
        ]
      + disabled                = (known after apply)
      + display_name            = (known after apply)
      + has_rsa_public_key      = (known after apply)
      + id                      = (known after apply)
      + login_name              = (known after apply)
      + name                    = "TEST_USER"
    }

Plan: 3 to add, 0 to change, 0 to destroy.

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
$ terraform apply 

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # snowflake_role.secondary_role will be created
  + resource "snowflake_role" "secondary_role" {
      + id   = (known after apply)
      + name = "SNOWFLAKE_SECOND_ROLE"
    }

  # snowflake_role.user_role will be created
  + resource "snowflake_role" "user_role" {
      + id   = (known after apply)
      + name = "SNOWFLAKE_USER_ROLE"
    }

  # snowflake_user.this[0] will be created
  + resource "snowflake_user" "this" {
      + default_role            = "SNOWFLAKE_USER_ROLE"
      + default_secondary_roles = [
          + "SNOWFLAKE_SECOND_ROLE",
        ]
      + disabled                = (known after apply)
      + display_name            = (known after apply)
      + has_rsa_public_key      = (known after apply)
      + id                      = (known after apply)
      + login_name              = (known after apply)
      + name                    = "TEST_USER"
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

snowflake_role.user_role: Creating...
snowflake_role.secondary_role: Creating...
snowflake_role.user_role: Creation complete after 1s [id=SNOWFLAKE_USER_ROLE]
snowflake_role.secondary_role: Creation complete after 1s [id=SNOWFLAKE_SECOND_ROLE]
snowflake_user.this[0]: Creating...
╷
│ Error: error creating user err = 001008 (22023): SQL compilation error:
│ invalid value [[SNOWFLAKE_SECOND_ROLE]] for parameter 'DEFAULT_SECONDARY_ROLES'
│ 
│   with snowflake_user.this[0],
│   on main.tf line 9, in resource "snowflake_user" "this":
│    9: resource "snowflake_user" "this" {
│ 
╵

```

**Additional context**

**Comment by mwiewior** [comment]
As discussed @dgniewek - it's due to the fact that the only supported value for `default_secondary_roles` is as  ('ALL') and I do agree that it should be documented more clearly in the provider documentation as well. I will create a PR shortly.

DEFAULT_SECONDARY_ROLES = ( 'ALL' )
Specifies the set of secondary roles that are active for the user’s session upon login. Secondary roles are a set of roles that authorize any SQL action other than the execution of CREATE <object> statements. The permissions to perform these actions can be granted to the primary role, secondary roles, or any lower roles in the role hierarchies.

Note that specifying a default secondary role for a user does not grant the role to the user. The role must also be granted explicitly to the user using the GRANT ROLE command.

The following values are supported:

ALL:
All roles that have been granted to the user.

Note that the set of roles is reevaluated when each SQL statement executes. If additional roles are granted to the user, and that user executes a new SQL statement, the newly granted roles are active secondary roles for the new SQL statement. The same logic applies to roles that are revoked from a user.

Default: NULL

**Comment by sfc-gh-swinkler** [comment]
merged #1584, will be released in next version. thanks @mwiewior for clarifying
